### PR TITLE
Additional restricted admin tests

### PR DESCRIPTION
**Pull request by anupama2501**
Issue:
https://github.com/rancher/qa-tasks/issues/164

Problem
Adding additional tests p1 cases to be covered for restricted admin.
Also fixed a few issues seen around rbac test cases.

Automated Testing
Jenkins job will be sent as part of the review.

### CODE REVIEW

File: tests/v2/validation/rbac/rbac.go [hunk 1]
```diff
@@ -170,3 +181,58 @@ func deleteLabels(labels map[string]string) {
...
+	config.LoadConfig(provisioning.ConfigurationFileKey, userConfig)
+
+	kubernetesVersion := userConfig.RKE1KubernetesVersions[0]
+	cni := userConfig.CNIs[0]
+	nodeProviders := userConfig.NodeProviders[0]
+
+	externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviders)
+
+	clusterConfig := ClusterConfig{nodeRoles: nodeAndRoles, externalNodeProvider: externalNodeProvider,
+		kubernetesVersion: kubernetesVersion, cni: cni}
```
Comment (markusewalker) [inline tests/v2/validation/rbac/rbac.go:235]: I'm assuming this is on a new line to not crowd line 234, is this correct?

**Comment by markusewalker** [comment]
Overall, lgtm, just had one question that I commented.

File: tests/v2/validation/rbac/rbac.go [hunk 2]
```diff
@@ -35,10 +38,18 @@ const (
 	psaWarn                 = "pod-security.kubernetes.io/warn"
 	psaAudit                = "pod-security.kubernetes.io/audit"
 	psaEnforce              = "pod-security.kubernetes.io/enforce"
+	kubeConfigTokenSetting = "kubeconfig-token-ttl-minutes"
```
Comment (izaac) [inline tests/v2/validation/rbac/rbac.go:41]: formatting , needs an extra space. `go.fmt`

File: tests/v2/validation/rbac/rbac_test.go [hunk 3]
```diff
@@ -206,12 +213,15 @@ func (rb *RBTestSuite) ValidateNS(role string) {
 func (rb *RBTestSuite) ValidateAddClusterRoles(role string) {
 
 	//Testcase7 Validate if project members are able to add other membes in cluster
```
Comment (izaac) [inline tests/v2/validation/rbac/rbac_test.go:215]: Typo in comment `members`

File: tests/v2/validation/rbac/rbac_test.go [hunk 4]
```diff
@@ -390,12 +406,35 @@ func (rb *RBTestSuite) ValidateCannotAddMPMsAsProjectOwner() {
...
+
+	assert.Equal(rb.T(), len(adminListSettings), len(resAdminListSettings))
+	assert.Equal(rb.T(), adminListSettings, resAdminListSettings)
+}
+
+func (rb *RBTestSuite) ValidateEditGlobalSettings() {
+	globalSettingByID, err := rb.steveStdUserclient.SteveType("management.cattle.io.setting").ByID(kubeConfigTokenSetting)
+	require.NoError(rb.T(), err)
+
+	updatedSettingValue, err := editGlobalSettings(rb.steveStdUserclient, globalSettingByID, "3")
```
Comment (izaac) [inline tests/v2/validation/rbac/rbac_test.go:430]: I can't tell which setting is this changing to `3`.

`ValidateEditGlobalSetting` is vague to show which setting is getting updated.

**Review by izaac — CHANGES_REQUESTED** [review]
A couple comments. This is looking good.

File: tests/v2/validation/rbac/rbac.go [hunk 5]
```diff
@@ -170,3 +181,58 @@ func deleteLabels(labels map[string]string) {
...
+	if err != nil {
+		return nil, err
+	}
+	return updateSetting, err
+}
+
+func listGlobalSettings(steveclient *v1.Client) (projectNames []string, err error) {
+	globalSettings, err := steveclient.SteveType("management.cattle.io.setting").List(nil)
+	if err != nil {
+		return nil, err
```
Comment (caliskanugur) [inline tests/v2/validation/rbac/rbac.go:197]: To expand my above explanation, if we were only using return here, we would send `projectNames` as `nil` slice and err with an assigned err value.

Please reach me offline if you have any questions regarding this.

File: tests/v2/validation/rbac/rbac_test.go (hunk 4 above)
Comment (caliskanugur) [reply tests/v2/validation/rbac/rbac_test.go:430]: Agreed I think maybe we can call the returned setting `kubeConfigTokenSetting` and its ID as `kubeConfigTokenSettingID`. Just to show an example please feel free to name however you see fit.

File: tests/v2/validation/rbac/rbac_test.go [hunk 6]
```diff
@@ -390,12 +406,35 @@ func (rb *RBTestSuite) ValidateCannotAddMPMsAsProjectOwner() {
...
+	require.Error(rb.T(), err)
+
+	k8sSetting, err := convertSetting(rb.steveStdUserclient, updatedSettingValue)
```
Comment (caliskanugur) [inline tests/v2/validation/rbac/rbac_test.go:433]: I've got the same nit here, I think we can call it `updatedXSetting`

**Review by caliskanugur — COMMENTED** [review]
I've got a few nits, looks good

File: tests/v2/validation/rbac/rbac.go (hunk 1 above)
Comment (anupama2501) [reply tests/v2/validation/rbac/rbac.go:235]: that is correct.

File: tests/v2/validation/rbac/rbac.go [hunk 7]
```diff
@@ -170,3 +181,58 @@ func deleteLabels(labels map[string]string) {
 		}
 	}
 }
+
+func convertSetting(steveAdminClient *v1.Client, globalSetting *v1.SteveAPIObject) (*v3.Setting, error) {
```
[the remark this answers is not in the document]
Comment (anupama2501) [reply tests/v2/validation/rbac/rbac.go]: You're right it was not needed. Might have been left from my test notes.

File: tests/v2/validation/rbac/rbac.go (hunk 5 above)
Comment (anupama2501) [reply tests/v2/validation/rbac/rbac.go]: Done. Couldn't resist in changing other functions with named return variables too :)

File: tests/v2/validation/rbac/rbac.go [hunk 8]
```diff
@@ -170,3 +181,58 @@ func deleteLabels(labels map[string]string) {
...
+	updateSetting := &v3.Setting{}
+	err := v1.ConvertToK8sType(globalSetting.JSONResp, updateSetting)
+	if err != nil {
+		return nil, err
+	}
+	return updateSetting, err
+}
+
+func listGlobalSettings(steveclient *v1.Client) (projectNames []string, err error) {
```
[the remark this answers is not in the document]
Comment (caliskanugur) [reply tests/v2/validation/rbac/rbac.go:194]: So, I don't think this function will work as it is. Probably a missed reassign and comment, `projectNames` isn't in use here.

**Review by caliskanugur — COMMENTED** [review]
There's only one unresolved comment, changes look good to me

File: tests/v2/validation/rbac/rbac.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (igomez06) [reply tests/v2/validation/rbac/rbac.go:194]: agreed

**Review by igomez06 — DISMISSED** [review]
Looks good. Anu is this being back ported?

**Review by markusewalker — APPROVED** [review]

**Review by caliskanugur — APPROVED** [review]
Looks good to me, could we have one last freeform offline?

**Review by izaac — DISMISSED** [review]
LGTM

**Review by markusewalker — COMMENTED** [review]
LGTM, @anupama2501 can you provide a a freeform job? I'll approve after.

**Review by igomez06 — APPROVED** [review]
LGTM

**Pull request closed without merging**